cmd/serve.go

```diff
@@ -79,32 +79,45 @@ func elementsHandler(config httpConfig, w http.ResponseWriter, r *http.Request)
 	path = strings.Replace(path, "/", ".", -1)
 	debug.Printf("Element path requested: %s", path)
 
-	config.EConfig.Path = path
-	elements := e.Elements{
-		Config: config.EConfig,
+  format := r.FormValue("format")
+	if format == "" {
+		format = config.OConfig.Format
 	}
 
+	config.OConfig.Format = format
```

Comment:
Two comments for the above change:

1. Line 82 is indented incorrectly. Go is odd with wanting 8-space hard-tabs, so misaligned lines are common for people just starting. I use `vi` and do the following two things to get around it:

a. Have the following in `~/.vimrc`:

```
set tabstop=2|set shiftwidth=2|set expandtab|set softtabstop=2
```

This translates all tabs/spaces into two-spaces for reading, and will then re-expand upon saving.

b. The `vim-go` plugin will automatically run `gofmt` upon saving the file.

If you don't use `vi`, you should run `gofmt` when you're ready to commit files. Most editors have plugins for `gofmt`, but it can be run on the shell manually:

```shell
$ gofmt path/to/file.go (stdout)
$ gofmt -w path/to/file.go (rewrites the file)
```

`gofmt` is a great built-in utility that has squashed all religious wars of tabs/spaces. `vim-go` is a great plugin that will also display in-line errors each time you `:w`, which saves a ton of time. It has a lot of extra features, but I only use it for auto-formatting + error display.

2. I'm pretty sure lines 82-87 can be reduced to:

```go
if v := r.FormValue("format"); v != "" {
  config.OConfig.Format = v             
}                                       
```

That should satisfy all conditions:

* Only set `config.OConfig.Format` if it's not an empty string
* Yield to `json` or `--format` if specified

But let me know if I missed something.

`v` is the accepted standard to use for throwaway or placeholder values. An exaggerated example of `v` (and scoping) can be seen [here](https://github.com/hashicorp/terraform/blob/master/builtin/providers/openstack/types.go#L121-L137) which tries to parse a JSON structure when the exact schema is unknown ahead of time.